### DISCUSSION TITLE
Vector cross-compilation for AIX POWER8

### DISCUSSION THREAD

**Discussion by mrgolangv** [vectordotdev/vector#9651] [Q&A] [answered]
Is it possible to cross-compile for the AIX OS?
There is very little information on the web about compiling rust for AIX (7.1).
What do you think about this? 
```
# uname -a
AIX

# lsconf | grep "Processor Type"
Processor Type: PowerPC_POWER8
```

**Comment by jszwedko** [comment, marked as answer]
Hi @mrgolangv !

Can you confirm if `powerpc64le-unknown-linux-gnu` would be the [target triple](https://wiki.osdev.org/Target_Triplet) for that platform? It does look like Rust has [tier 2 support](https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-2-with-host-tools) for that platform. I'll try building Vector with it and see what happens.

**Reply by jszwedko** [reply]
Unfortunately it looks like `ring`, one of our transitive dependencies, does not yet support PowerPC as a target: https://github.com/briansmith/ring/issues/389 . That would need to be resolved first.

**Reply by mrgolangv** [reply]
Hi @jszwedko !

I think powerpc64le-unknown-linux-gnu is not suitable for AIX OS.
A simple program compiled for powerpc64le-unknown-linux-gnu cannot be run on AIX.

Original executable file for AIX OS:

```
#file ../../echo
../../echo: executable (RISC System/6000 V3.1) or obj modul


#file /usr/bin/zabbix_sender (C lang)
/usr/bin/zabbix_sender: 64-bit XCOFF executable or object module not stripped
```
Compiled for powerpc64le-unknown-linux-gnu:
```
file target/powerpc64le-unknown-linux-gnu/debug/foobar
target/powerpc64le-unknown-linux-gnu/debug/foobar: ELF 64-bit LSB executable, 64-bit PowerPC or cisco 7500, version 1 (GNU/Linux), statically linked, for GNU/Linux 3.2.0, BuildID[sha1]=8ed7faf46f9d8da6d26924559a55c364cd3b1c86, with debug_info, not stripped
```
Yes, I also encountered ring compilation errors.
It would be cool to run the vector for the AIX and Solaris (Sun OS).

Thanks for responding back.

**Reply by jszwedko** [reply]
Aha, I see, that does seem different. Could you try running `gcc -dumpmachine`? I think that will spit out the target triplet for the platform.
